packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -236,7 +239,8 @@ export class RunningSummarizer implements IDisposable {
         logger: ITelemetryLogger,
         summaryWatcher: IClientSummaryWatcher,
         configuration: ISummaryConfiguration,
-        generateSummary: (full: boolean, safe: boolean) => Promise<GenerateSummaryData | undefined>,
+        // eslint-disable-next-line max-len
+        generateSummary: (full: boolean, safe: boolean, summaryLogger: ITelemetryLogger) => Promise<GenerateSummaryData | undefined>,
```

Comment:
nit: here, and below: I think summaryLogger is too verbose. logger is good enough, there is no need to encode context (class name) into arg names

Reply:
summaryLogger isn't a class name - it's an important distinction, that the context of this logger is specific to the particular summar**y** (not summar**izer**).  I'm thinking about renaming this and the field in the logs to be about "runs", to leverage the terminology as in "RunningSummarizer".  Not sold on it though, still kind of verbose/potentially confusing.  #NamingIsHard